onnx/test/inference_function_test.py

```diff
@@ -145,6 +152,47 @@ def test_reshape_inference(self) -> None:
             {"t": np.array([2, 2, 5], dtype=np.int64)},
         ) == _to_tensor_types({"y": (onnx.TensorProto.FLOAT, (2, 2, 5))})
 
+    def test_scan_inference_with_subgraph(self) -> None:
+        seq_len = "sequence"
+        input_size = 2
+        loop_state_size = 3
+
+        input_value_infos = [
+            make_tensor_value_info("loop_state_in", TensorProto.UNDEFINED, None),
+            make_tensor_value_info("input", TensorProto.UNDEFINED, None),
+        ]
+        output_value_infos = [
+            make_tensor_value_info("loop_state_out", TensorProto.UNDEFINED, None),
+            make_tensor_value_info("output", TensorProto.UNDEFINED, None),
+        ]
+
+        subgraph = make_graph(
+            [
+                make_node("Identity", ["loop_state_in"], ["loop_state_out"]),
+                make_node("Identity", ["input"], ["output"]),
+            ],
+            "subgraph",
+            input_value_infos,
+            output_value_infos,
+        )
+
+        assert onnx.shape_inference.infer_node_outputs(
+            onnx.defs.get_schema("Scan", 9),
+            make_node(
+                "Scan",
+                ["loop_state_orig", "scan_input"],
+                ["loop_state_final", "scan_output"],
+                num_scan_inputs=1,
+                body=subgraph,
+            ),
```

Comment:
FYI: I didn't use parser for creating NodeProto here, because it seems that parse_node is missing.